### PR TITLE
Add missing includes

### PR DESCRIPTION
**Pull request by Tectu**
Adding a bunch of missing includes.
These were spotted while updating the FreeBSD port.

## What type of PR is this? (check all applicable)

* [ ] Refactor
* [ ] Feature
* [ ] Bug Fix
* [ ] Optimization
* [ ] Documentation Update
* [x] Other: *Fix build*

## Description
No user facing changes. Just fixing the build.

## Related Issues & Documents
N/A

## Documentation (check all applicable)

* [ ] This PR requires changes to the Wiki documentation (describe the changes)
* [ ] This PR requires changes to the documentation inside the git repo (please add them to the PR).
* [x] Does not require documentation changes

### CODE REVIEW

File: src/ipc/ipc.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "ipc/ipc.hpp"
 
 #include <sys/stat.h>
+#include <sys/wait.h>
```
Comment (patrick96) [inline src/ipc/ipc.cpp:4]: Why is this needed? I don't see any wait calls in that file

File: include/cairo/context.hpp [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #include <cairo/cairo-xcb.h>
 
 #include <algorithm>
+#include <array>
```
Comment (patrick96) [inline include/cairo/context.hpp:6]: The missing `<array>` header is already in the hotfix branch, see #3038. Should also be on master within the week

File: src/ipc/ipc.cpp (hunk 1 above)
Comment (Tectu) [reply src/ipc/ipc.cpp:4]: It's missing the definition for `WEXITSTATUS` (used in line 272).

File: include/cairo/context.hpp (hunk 2 above)
Comment (Tectu) [reply include/cairo/context.hpp:6]: Alright!
Shall I update the PR to exclude this?

File: src/ipc/ipc.cpp (hunk 1 above)
Comment (patrick96) [reply src/ipc/ipc.cpp:4]: This file only has 221 lines

File: include/cairo/context.hpp (hunk 2 above)
Comment (patrick96) [reply include/cairo/context.hpp:6]: Please remove them. My other PR adds them directly into `common.hpp`

File: src/ipc/ipc.cpp (hunk 1 above)
Comment (Tectu) [reply src/ipc/ipc.cpp:4]: Aye... this was for `lib/i3ipcpp/src/ipc.cpp`.
I'll re-create this PR after I got some much needed sleep...

**Comment by patrick96** [comment]
I added the missing header in polybar/i3ipcpp#16 and #3052
